crates/bevy_ecs/src/system/system_param.rs

```diff
@@ -2692,12 +2645,43 @@ unsafe impl SystemParam for FilteredResourcesMut<'_, '_> {
     }
 }
 
-/// An error that occurs when a system parameter is not valid.
+/// An error that occurs when a system parameter is not valid,
+/// used by system executors to determine what to do with a system.
 ///
-/// Generated when [`SystemParam::validate_param`] returns `false`,
+/// Returned as an error from [`SystemParam::validate_param`],
 /// and handled using the unified error handling mechanisms defined in [`bevy_ecs::error`].
 #[derive(Debug, PartialEq, Eq, Clone, Display, Error)]
-pub struct SystemParamValidationError;
+pub struct SystemParamValidationError {
+    /// Whether the system should be skipped.
+    pub skipped: bool,
```

Comment:
Do we need this to be public? If we're already exposing `is_skipped`, etc, I don't see the need for this. Making it private would make this easier to change in the future, and it forces users to use the better documented functions.
```suggestion
    skipped: bool,
```

Reply:
I'd prefer to keep this pub and kill the helper method if we're deduplicating.

Reply:
Whoops, I had started with it `pub` and failed to remove it after I added the method.  Good catch!  

But it sounds like we want it `pub`, so I'll go remove the helper methods instead.  (I guess I still have some old C# habits.)

Reply:
Works for me!